x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -30,41 +30,49 @@
-                    IndexSettings.LOOK_AHEAD_TIME.getKey(),
-                    IndexSettings.LOOK_AHEAD_TIME.getDefault(allSettings)
-                );
+            DataStream dataStream = metadata.dataStreams().get(dataStreamName);
+            IndexMode indexMode;
+            if (dataStream != null) {
+                indexMode = dataStream.getIndexMode();
+            } else {
+                indexMode = matchingTemplate.getDataStreamTemplate().getIndexMode();
+            }
+            if (indexMode != null) {
                 Settings.Builder builder = Settings.builder();
-                DataStream dataStream = metadata.dataStreams().get(dataStreamName);
-                Instant start;
-                if (dataStream == null) {
-                    start = Instant.ofEpochMilli(resolvedAt).minusMillis(lookAheadTime.getMillis());
-                } else {
-                    IndexMetadata currentLatestBackingIndex = metadata.index(dataStream.getWriteIndex());
-                    if (currentLatestBackingIndex.getSettings().hasValue(IndexSettings.TIME_SERIES_END_TIME.getKey()) == false) {
-                        throw new IllegalStateException(
-                            String.format(
-                                Locale.ROOT,
-                                "backing index [%s] in tsdb mode doesn't have the [%s] index setting",
-                                currentLatestBackingIndex.getIndex().getName(),
-                                IndexSettings.TIME_SERIES_START_TIME.getKey()
-                            )
-                        );
+                builder.put(IndexSettings.MODE.getKey(), indexMode.name().toLowerCase(Locale.ROOT));
+
+                if (indexMode == IndexMode.TIME_SERIES) {
+                    TimeValue lookAheadTime = allSettings.getAsTime(
+                        IndexSettings.LOOK_AHEAD_TIME.getKey(),
+                        IndexSettings.LOOK_AHEAD_TIME.getDefault(allSettings)
+                    );
+                    Instant start;
+                    if (dataStream == null) {
+                        start = Instant.ofEpochMilli(resolvedAt).minusMillis(lookAheadTime.getMillis());
+                    } else {
+                        IndexMetadata currentLatestBackingIndex = metadata.index(dataStream.getWriteIndex());
+                        if (currentLatestBackingIndex.getSettings().hasValue(IndexSettings.TIME_SERIES_END_TIME.getKey()) == false) {
+                            throw new IllegalStateException(
+                                String.format(
+                                    Locale.ROOT,
+                                    "backing index [%s] in tsdb mode doesn't have the [%s] index setting",
+                                    currentLatestBackingIndex.getIndex().getName(),
+                                    IndexSettings.TIME_SERIES_START_TIME.getKey()
+                                )
+                            );
```

Comment:
I think this will throw an exception when someone converts a data stream from "standard" to "time_series" mode right? Because the current backing index wouldn't have the end time, but the new one will look for the prior index's end date?

(It might not be something we're trying to fix for right, just trying to understand the behavior)

Reply:
> I think this will throw an exception when someone converts a data stream from "standard" to "time_series" mode right?

Yes, it will fail. I think we should address migrating a regular data stream into a tsdb data stream in a separate change and tweaking this validation logic I think is part of that.